### PR TITLE
Add the {:one_of, choices} type

### LINKED ISSUE #15: Member type
A pretty common possible option value is a member of a list. An example that comes to mind is Broadway's `:batch_mode` in `Broadway.test_messages/3`, which is `:bulk` or `:flush`.

Do we want to support a "member" type?

```elixir
schema = [
  batch_mode: [type: {:member, [:bulk, :flush]}, default: :flush]
]
```

Do you think it's too soon, too specific, or a good addition?

### PR DESCRIPTION
**Pull request by whatyouhide**
Closes #15.

### CODE REVIEW

File: lib/nimble_options.ex [hunk 1]
```diff
@@ -93,6 +93,8 @@ defmodule NimbleOptions do
 
     * `{:fun, arity}` - Any function with the specified arity.
 
+    * `{:one_of, choices}` - A value that is a member of one of the `choices`.
```
Comment (josevalim) [inline lib/nimble_options.ex:96]: Reading the docs... now I am thinking `:choices` would be the better name? :D

Comment (whatyouhide) [reply lib/nimble_options.ex:96]: @josevalim I think `{:choices, ...}` looks weird when used in practice. `{:one_of, [:flush, :bulk]}` reads better to me than `{:choices, [:flush, :bulk]}`. I don't have a strong preference though, so just tell me what you prefer and I'll implement that :)

Comment (wojtekmach) [reply lib/nimble_options.ex:96]: agreed about :one_of. another suggestion would be `{:in, [:flush, :bulk]}`? but not strong opinion either.

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**